apps/web/src/views/Swap/V3Swap/components/SwapModalFooter.tsx

```diff
@@ -25,30 +29,53 @@ export const SwapModalFooter = memo(function SwapModalFooter({
   lpFee: realizedLPFee,
   inputAmount,
   outputAmount,
+  trade,
   tradeType,
-  executionPrice,
   slippageAdjustedAmounts,
   isEnoughInputBalance,
   onConfirm,
   swapErrorMessage,
   disabledConfirm,
+  isMM,
+  isRFQReady,
+  currencyBalances,
 }: {
+  trade: SmartRouterTrade<TradeType> | undefined
   tradeType: TradeType
   lpFee: CurrencyAmount<Currency>
   inputAmount: CurrencyAmount<Currency>
   outputAmount: CurrencyAmount<Currency>
   priceImpact: Percent
-  executionPrice: Price<Currency, Currency>
   slippageAdjustedAmounts: { [field in Field]?: CurrencyAmount<Currency> }
   isEnoughInputBalance: boolean
-  onConfirm: () => void
   swapErrorMessage?: string | undefined
   disabledConfirm: boolean
+  isMM?: boolean
```

Comment:
actually it's good to separate MM and AMM as mixing them might be difficult to maintain in the future

Reply:
we can keep it this way for now and refactor later though